### ISSUE TITLE
Building on FreeBSD

### ISSUE THREAD

**Issue by geier** [Valloric/ycmd#500]
Since https://github.com/Valloric/ycmd/commit/15f3d0443112d1aa488d339ae5134059f74ba7ce ycmd (and with that YouCompleteMe) refuses to build on FreeBSD, but `FindPythonLibrariesOnLinux()` also works on FreeBSD (for me).

As YouCompleteMe claims to work on FreeBSD I see two options: either remove that claim or apply something like the following patch (as a FreeBSD user I obviously prefer the latter option):

```diff
diff --git a/build.py b/build.py
index 5ab86d2..48397c1 100755
--- a/build.py
+++ b/build.py
@@ -53,7 +53,7 @@ LIBRARY_LDCONFIG_REGEX = re.compile(
 
 
 def OnLinux():
-  return platform.system() == 'Linux'
+  return platform.system() in ['Linux', 'FreeBSD']
 
 
 def OnMac():
```

**Comment by vheon** [comment]
We're already working on the installer #499

**Comment by micbou** [comment]
> As YouCompleteMe claims to work on FreeBSD I see two options: either remove that claim or apply something like the following patch (as a FreeBSD user I obviously prefer the latter option):

We don't claim that. From the [README](https://github.com/Valloric/YouCompleteMe#freebsdopenbsd):
> OpenBSD / FreeBSD are not officially supported platforms by YCM.

Also, the `build.py` script not working on FreeBSD and YCM not working on it are two different things: if the script fails, you can still [build YCM by following the full instructions](https://github.com/Valloric/YouCompleteMe#full-installation-guide).

Anyway, this should be fixed by PR #499.

**Comment by geier** [comment]
You are right and I worded what I meant poorly. My point is that with the current install script it will just refuse to work on anything but Linux, OS X and Windows, but the TODO gives instructions for Free and OpenBSD (with a note of not supporting them) which cannot currently work.

Sorry for the noise and thanks for great software!
